src/containers/Login/Login.js

```diff
@@ -63,32 +63,34 @@ export class Login extends Component {
     this.handleUser(data)
   }
 
-  handleUser = (data) => {
+  handleUser = async (data) => {
     const validUser = data.api_key
     const userRole = data.role
-    this.props.setUser(validUser)
-    this.props.setRole(userRole)
-    this.handleRedirect()
+    await this.props.setUser(validUser)
+    await this.props.setRole(userRole)
+    await this.handleRedirect()
   }
 
   handleRedirect = () => {
     if (this.props.role === 'Student') {
       this.props.history.push('/student-dashboard')
-    } else {
+    } else if (this.props.role === 'Instructor') {
       this.props.history.push('/dashboard')
+    } else {
+      console.log("NO LOGIN")
```

Comment:
We will want to handle this condition another way. We may not need it at all because each login sets a user as either Instructor or Student